### ISSUE TITLE
The library doesn't use Levenshtein at all, or at least it's inconsistent

### ISSUE THREAD

**Issue by R-N** [seatgeek/thefuzz#53]
This library is described as fuzzy string matching with Levenshtein distance. However, it doesn't seem to use Levenshtein at all?

fuzz.ratio("tide", "diet") returns:
- 0.5 with python-Levenshtein installed
- 0.25 without python-Levenshtein

This is because **python-Levenshtein uses Indel distance for ratio instead of Levenshtein**. [Issue 47](https://github.com/maxbachmann/Levenshtein/issues/47) in python-Levenshtein mentioned that [Levenshtein.ratio is based upon the Indel distance and not on the Levenshtein distance](https://github.com/maxbachmann/Levenshtein/issues/47#issuecomment-1412060377). [(That's the correct repo I think)](https://github.com/seatgeek/thefuzz/issues/37). Indel distance is similar to Levenshtein, but they are not the same and may produce different results.

But it doesn't seem to just end there. **difflib, the fallback for when you don't have python-Levenshtein installed, doesn't use Levenshtein distance either**. [Issue 128](https://github.com/seatgeek/fuzzywuzzy/issues/128) in the older repository/version of this package mentioned that difflib uses ["Ratcliff-Obershelp algorithm, not the Levenshtein distance"](https://github.com/seatgeek/fuzzywuzzy/issues/128#issuecomment-766208348). 

TL;DR:
- This library doesn't seem to use Levenshtein distance at all.
- Installing python-Levenshtein doesn't just simply give speed up. It will produce different results because different algorithms are used.

Note: This library uses SequenceMatcher.ratio for all of its function with ratio in its name, either from Levenshtein package or difflib.

**Comment by maxbachmann** [comment]
Note that the Indel distance is the same as the Levenshtein distance with substitutions weighted as 2. Still I agree this is misleading. I think to a part this stems from python-Levenshtein, where `Levenshtein.distance` calculates the Levenshtein distance, but `Levenshtein.ratio` calculates the normalized Indel similarity (In my defense this was implemented before I became the maintainer of `python-Levenshtein` :sweat_smile: )

**Comment by R-N** [comment]
> Note that the Indel distance is the same as the Levenshtein distance with substitutions weighted as 2. 

Yeah, I also mentioned they're similar. Still, they're not the same.

> I think to a part this stems from python-Levenshtein, where `Levenshtein.distance` calculates the Levenshtein distance, but `Levenshtein.ratio` calculates the normalized Indel similarity

Yup. I mentioned that too. I can't help but wonder why. The library is literally named Levenshtein. It could've been an option instead of the default, having "Indel" in the function name like it does for Jaro, or as an optional parameter.

Well, I think the readme should at least be changed to tell people about this. It's where they'd read first and it says "It uses [Levenshtein Distance](https://en.wikipedia.org/wiki/Levenshtein_distance)" at the very beginning.

**Comment by cornzz** [comment]
This is really confusing to me, I also stumbled upon the fact that different ratios are returned depending on whether python-Levenshtein is installed or not. What the hell, how is this acceptable?

**Comment by maxbachmann** [comment]
In thefuzz this is no longer the case for 1.5 years. Since https://github.com/seatgeek/thefuzz/pull/10 it always uses the same implementation. The basis for ratio is the Levenshtein distance with a substitution weight of 2. This is also known as InDel distance.
